index.js

```diff
@@ -147,10 +148,19 @@ RedirectableRequest.prototype.removeHeader = function (name) {
   this._currentRequest.removeHeader(name);
 };
 
+// Re-bind timeout handler to all subsequent requests
+RedirectableRequest.prototype.setTimeout = function (msecs, callback) {
+  this._timeoutMsecs = msecs;
+  if (callback) {
+    this.once("timeout", callback);
+  }
+  return this._currentRequest.setTimeout(msecs);
```

Comment:
We might want to mirror [this code](https://github.com/nodejs/node/blob/v11.9.0/lib/_http_outgoing.js#L182-L193), except that we use the global function `setTimeout` instead of `socket.setTimeout`, storing the return value as `this._timeout` or so. When we reach the timeout, the current request should be aborted (without emitting the abort event) and the timeout event should be fired.
If the request succeeds before timeout, the timeout handler should be canceled.

If a new timeout value is set, we should clear the previous one.

Reply:
Ahh... okay, I follow now.